onnxruntime/test/ir/onnx_model_test.cc

```diff
@@ -187,5 +189,152 @@ TEST_F(ONNXModelsTest, TestModelsWithAnOpContainingAFunctionBody) {
   ASSERT_STATUS_OK(model->MainGraph().Resolve());
 }
 
+TEST(FunctionVerification, VerifyModelLocalFunctions) {
+  const char* code = R"ONNX(
```

Comment:
would be good to describe what exactly is being tested. is it based support for model local functions? does anything here validate name replacement when say multiple nodes use the same model local function? etc.